src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -1016,19 +1016,37 @@ def on_demand_eps_snql_factory(
     return rate_snql_factory(aggregate_filter, interval, 1, alias)
 
 
+# Compare with discover:
+# https://github.com/getsentry/sentry/blob/4dc9c8b46538c1c4d8397344c9ccaf6ad8fd8d52/src/sentry/search/events/datasets/discover.py#L1506-L1553
 def on_demand_user_misery_snql_factory(
     aggregate_filter: Function, org_id: int, use_case_id: UseCaseID, alias: str | None = None
 ) -> Function:
-    miserable_users = uniq_if_column_snql(
-        aggregate_filter, org_id, use_case_id, "satisfaction", "frustrated"
+    # XXX: To my suprise, this is returning 6 miserable users rather than 4 users
+    # I may be setting things up incorrectly in the test
+    _miserable_users_new = miserable_users(org_id, [aggregate_filter.parameters[1]])
```

Comment:
Using the function `miserable_users` does work. There's a tiny difference but it does the work.
I still get the same test results, thus, convincing me that I'm not setting up the tests correctly.

Notice that I'm basically now using the passed `aggregate_filter` which is obviously not right.